src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -8295,6 +8322,8 @@ extern "C" bool QCALLTYPE SfiInit(StackFrameIterator* pThis, CONTEXT* pStackwalk
     {
         GCX_COOP();
         UpdatePerformanceMetrics(&pThis->m_crawl, false, ((uint8_t)pExInfo->m_kind & (uint8_t)ExKind::RethrowFlag) == 0);
+
+        FailFastIfCorruptingStateException(pExInfo);
```

Comment:
I am wondering whether it would be possible to do this earlier. The earlier we failfast for corrupting exceptions, the better.

Reply:
We may be able move it to the beginning of the SfiInit, but other than that, this method is the first native code that gets called when we have the throwable so that I can get a proper message out of it.